### PR TITLE
Project : Bare Metal Forms and Helpers

### PR DESCRIPTION
**Pull request by marcoshdezcam**
### 
We created a form to create and edit a user using plain HTML and Rails helpers (#form_tag and #form_for). 

### 👤 Authors
@marcoshdezcam and @che30 

### 🔧 Built With
- RoR
- Gitflow Workflow
- Rubocop
- SASS

Thank you for taking the time to review our code! 💐

### CODE REVIEW

File: app/views/users/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<%
+=begin %>
+ <form action="/users" method="POST" accept-charset="utf-8">
```
Comment (teekaytech) [inline app/views/users/_form.html.erb]: Change this to `action="users_path"` helper action, and not `action="/users"`

File: app/views/users/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+   <label for="username">Username:</label>
```
Comment (teekaytech) [inline app/views/users/_form.html.erb]: Use 
```html
  <label for="user[username]">Username</label>
```

File: app/views/users/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+<div id="error_explanation">
+<h2><%= pluralize(@user.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+<ul>
+    <% @user.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+    <% end %>
+</ul>
+</div>
+<% end %>
```
Comment (teekaytech) [inline app/views/users/_form.html.erb]: Move this logic to a helper file.

File: app/views/users/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+<div>
+    <div>
+        Your username here:<br>
+        <%= f.text_field :username, value: "che & marcos" %>
+    </div>
+    <div>
+        Your email here:<br>
```
Comment (teekaytech) [inline app/views/users/_form.html.erb]: Use rails label helper: `<%= u.label :email %>`

File: app/views/users/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+<ul>
+    <% @user.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+    <% end %>
+</ul>
+</div>
+<% end %>
+<div>
+    <div>
+        Your username here:<br>
```
Comment (teekaytech) [inline app/views/users/_form.html.erb]: Use rails label helper: <%= u.label :email %>

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've done really well in this project.
You just have to make some required changes before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] Update your README.md file. Important sections (like project setup, description, etc) are missing. 

- [ ]  There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.
[Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
[The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- Check my inline comments below.


**[ OPTIONAL CHANGES ]**

- [ ] You can validate the email format by using either a regular expression or a built-in URI method. [This STO thread](https://stackoverflow.com/questions/38611405/email-validation-in-ruby-on-rails) might help.


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by che30** [comment]
Implemented all changes as requested by the TSE

File: app/views/users/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+<div id="error_explanation">
+<h2><%= pluralize(@user.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+<ul>
+    <% @user.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+    <% end %>
+</ul>
+</div>
+<% end %>
```
Comment (acushlakoncept) [inline app/views/users/_form.html.erb]: - [ ] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it. **The Resources below may come in handy**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Great job! 👏  Both of you have done well on this project.
I see you have implemented almost all the changes requested by the previous TSE, 
Let's quick fix this last one you missed and request another review then you should be done.

See comments below:


### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

File: app/helpers/users_helper.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+module UsersHelper
+  def show_errors(_user)
+    if @user.errors.any?
+      puts 'error_explanation'
+      puts pluralize(@user.errors.count, 'error')
+      print prohibited this post from being saved:
```
Comment (ghost) [inline app/helpers/users_helper.rb]: On line six, the statement after print should be in quotes since it is not a variable.

**Review by ghost — APPROVED** [review]
# Status: Approved 🎉 

**Team, before you merge please check the comment below. Otherwise, congratulations and all the best in your next project 🥂.**

**Pull request merged**